torch/_inductor/scheduler.py

```diff
@@ -806,42 +889,58 @@ def fuse_nodes_once(self):
             - self.score_fusion(): assigns priority to a given fusion
         """
         fused_nodes = set(self.nodes)
+
+        # recent_fusions tracks names of nodes fused during this function call.
+        #
+        # This is required because config.fusion_reorder_loops, when enabled,
+        # causes get_possible_fusions() to return new nodes not in
+        # name_to_fused_node, so we can't assume
+        #   node = self.name_to_fused_node[node.get_first_name()]
+        # will map an unfused nodes back to itself
+        recent_fusions: Set[str] = set()
```

Comment:
Why can't we update `self.name_to_fused_node` properly?  I don't totally get why this is needed.

Reply:
There isn't a 1-1 mapping between name and nodes, so it's impossible to represent in `name_to_fused_nodes`. We also don't want to, since the reordered nodes only exist for the duration of this function call.